core/src/main/java/org/jivesoftware/spark/component/JContactItemField.java

```diff
@@ -176,7 +177,11 @@ private void showPopupMenu() {
 
         if (validItems.size() != 0 && !popup.isVisible()) {
             popup.pack();
-            popup.setSize(textField.getWidth(), 200);
+            if(validItems.size()*16<=SparkManager.getMainWindow().getHeight()-SparkManager.getSearchManager().getSearchServiceUI().getHeight()-SparkManager.getMainWindow().getTopToolBar().getHeight()-SparkManager.getWorkspace().getStatusBar().getHeight()-48){
```

Comment:
For future reference: lets not make lines this long, because it makes it hard to read what's going on.